tests/test_misc.py

```diff
@@ -2789,6 +2789,59 @@ def test_emergencyrecover(node_factory, bitcoind):
     assert l2.rpc.listfunds()["channels"][0]["state"] == "ONCHAIN"
 
 
+@unittest.skipIf(os.getenv('TEST_DB_PROVIDER', 'sqlite3') != 'sqlite3', "sqlite3-specific DB rollback")
+@pytest.mark.openchannel('v1')
+@pytest.mark.openchannel('v2')
+def test_recover_plugin(node_factory, bitcoind):
+    l1 = node_factory.get_node(may_reconnect=True, options={'log-level': 'info', 'experimental-peer-storage': None},
+                               allow_warning=True,
+                               feerates=(7500, 7500, 7500, 7500))
+    l2 = node_factory.get_node(may_reconnect=True, options={'log-level': 'info', 'experimental-peer-storage': None},
+                               feerates=(7500, 7500, 7500, 7500), allow_broken_log=True, allow_bad_gossip=True)
+
+    l3 = node_factory.get_node(may_reconnect=True, options={'log-level': 'info', 'experimental-peer-storage': None},
+                               feerates=(7500, 7500, 7500, 7500), allow_broken_log=True)
+
+    l1.rpc.connect(l2.info['id'], 'localhost', l2.port)
+
+    l1.fundchannel(l2, 10**6)
+    mine_funding_to_announce(bitcoind, [l1, l2, l3])
+
+    l2.stop()
+
+    # Save copy of the db.
+    dbpath = os.path.join(l2.daemon.lightning_dir, TEST_NETWORK, "lightningd.sqlite3")
+    orig_db = open(dbpath, "rb").read()
+    l2.start()
+
+    l2.rpc.connect(l3.info['id'], 'localhost', l3.port)
+
+    l2.fundchannel(l3, 10**6)
+    mine_funding_to_announce(bitcoind, [l1, l2, l3])
+
+    l1.pay(l2, 200000000)
+
+    l2.restart()
+
+    # Now, move l2 back in time.
+    l2.stop()
+
+    # Overwrite with OLD db.
+    open(dbpath, "wb").write(orig_db)
+    l2.start()
+
+    bitcoind.generate_block(5, wait_for_mempool=1)
+    sync_blockheight(bitcoind, [l1, l2, l3])
+
+    # l2.daemon.wait_for_log(r'All outputs resolved.*')
+    wait_for(lambda: l2.rpc.listfunds()["channels"][0]["state"] == "ONCHAIN")
+    wait_for(lambda: l2.rpc.listfunds()["channels"][1]["state"] == "ONCHAIN")
```

Comment:
CI failing here: seems like l2 only has one channel?

Reply:
It passes locally, but fails here for some reason. I am trying to resolve it...